src/components/shared/thread/ActivityGraph.js

```diff
@@ -80,7 +87,7 @@ export class ThreadActivityGraph extends React.PureComponent<Props, State> {
     }
     const rect = canvas.getBoundingClientRect();
     this.setState({
-      hoveredSample: this._getSampleAtMouseEvent(event),
+      hoveredPixelState: this._getSampleAtMouseEvent(event),
```

Comment:
I'm a bit concerned about this: indeed `hoveredPixelState` will change at every call, so the state changes at each mousemove, which will trigger a rerender even if the values don't change.

This isn't a big issue in itself, because in `render` we explode this. So we'd just run `render` some more, that's probably OK.

__However__ the same happens for `cpuRatioInTimeRange` inside `hoveredPixelState`, and this means we rerender the full `SampleTooltipContents` at each mousemove even when nothing changes. The result is that the React diffing algorithm will need to run, and this is somewhat costly.
The way to fix this would be to extract the 2 parts of SampleTooltipContents (CPU / the rest) to separate components, so that only the CPU component would need to rerender when cpuRatioInTimeRange changes, which is much smaller (only one line).
Another easy (maybe easier) way is to take the 2 values of `cpuRatioInTimeRange` as separate props, so that the compare function could check them just fine.
Also both approaches can be combined.
I'll let you choose :-)